datashape/coretypes.py

```diff
@@ -87,7 +87,9 @@ def info(self):
         return type(self), self.parameters
 
     def __eq__(self, other):
-        return type(self) == type(other) and self.info() == other.info()
+        return (issubclass(type(other), Mono) and
+                self.shape == other.shape and
+                self.measure.info() == other.measure.info())
```

Comment:
Does the `__hash__` need to be updated at all, or is `self.info()` still sufficient for equal-comparing instances to hash equal?